activerecord/lib/active_record/model_schema.rb

```diff
@@ -432,8 +432,13 @@ def columns
       end
 
       def _returning_columns_for_insert # :nodoc:
-        @_returning_columns_for_insert ||= columns.filter_map do |c|
-          c.name if connection.return_value_after_insert?(c)
+        @_returning_columns_for_insert ||= columns.filter_map do |column|
+          column_name = column.name
+          if connection.return_value_after_insert?(column) ||
+              (composite_primary_key? && primary_key.include?(column_name)) ||
```

Comment:
Unless I'm missing some corner-case `columns` array should always include primary key column which means the condition will be true at least for one column in the list.
I think instead of making the check we should always push the `primary_key` value to the list and make sure the resulting list is unique

Reply:
Yes, good idea 👍

Reply:
@nvasilevski I'd say it's good enough to special case for a single-column PK?